### PR TITLE
Move `copyImg` to `ImageManager` and make reusable

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Move `copyImg` to `ImageManager` to make it reusable.
| Type?             |  refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A.
| Related PRs       | N/A.
| How to test?      | Make sure that product import still work with images.
| Possible impacts? | Import with images link.

## 1 - Try the code bellow
```php

<?php

require (dirname(__FILE__) . '/config/config.inc.php');

function setProducCover(int $idProduct, string $url)
{
      $image = new Image();
      $image->id_product = $idProduct;
      $image->position = Image::getHighestPosition($idProduct) + 1;
      $image->cover =  1;
        
      if(!$image->add()) {
          return false;
      }

      if (!ImageManager::copyImg($idProduct, $image->id, $url, 'products', true)) {
          $image->delete();
          return false;
      }

      return true;
}

$idProduct = 10 ; // Adapt please
$imageUrl = 'https://picsum.photos/536/354';
setProducCover($idProduct, $imageUrl);

```

### CODE REVIEW

File: controllers/admin/AdminImportController.php
Comment (matks) [inline controllers/admin/AdminImportController.php]: Too many dollar signs 😄 I guess

**Review by matks — CHANGES_REQUESTED** [review]

File: controllers/admin/AdminImportController.php
Comment (PululuK) [reply controllers/admin/AdminImportController.php]: Money money money 😄 
Nice catch @matks

**Review by matks — APPROVED** [review]

**Review by Amoifr — COMMENTED** [review]
You're right these have no place in controller. And as you said they will reusable in this place. Thx

**Comment by PrestaEdit** [comment]
Maybe we could deprecated theses two methods, there is no needs to call a method that call another one, especially while it's static: best will be to use it directly (even in the existing code, if possible).

**Comment by matks** [comment]
@PrestaShop/committers I approved the PR but it is missing 1 approval :) if you want to look at it

**Review by zuk3975 — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by FabienPapet — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @PululuK 

Thank you for your PR, I tested it and it seems to works, as we can see here :

[Untitled_ Nov 16, 2022 5_01 PM.webm](https://user-images.githubusercontent.com/100699445/202233127-f4440556-7a69-413b-852f-4ee6d65aa466.webm)

Can you give me an example of module to let us test the extra code you provided? 

Waiting for your feedback

**Comment by AureRita** [comment]
hi @PululuK 

After checking with you, I have create a php files with the code you've send and replace the id by a product without picture, 
I tested it and it works, as we can see : 

[Untitled_ Nov 18, 2022 5_07 PM.webm](https://user-images.githubusercontent.com/100699445/202749818-a4b0e483-eb3b-417b-aef9-46321dffde1e.webm)

just a little disclaimer, the link of the picture you send me don't work so I use another one

Thank you

**Review by AureRita — APPROVED** [review]

**Comment by matks** [comment]
Thank you @PululuK

**Pull request merged**
